### PR TITLE
Add enum forward map for go

### PR DESCRIPTION
**Pull request by mlsteele**
For an avdl enum you now not only get this

```
type ProofState int

const (
	ProofState_NONE         ProofState = 0
	ProofState_OK           ProofState = 1
	ProofState_TEMP_FAILURE ProofState = 2
	ProofState_PERM_FAILURE ProofState = 3
	ProofState_LOOKING      ProofState = 4
	ProofState_SUPERSEDED   ProofState = 5
	ProofState_POSTED       ProofState = 6
	ProofState_REVOKED      ProofState = 7
)
```

but also this!

```
var ProofState_map = map[string]ProofState{
	"NONE":         0,
	"OK":           1,
	"TEMP_FAILURE": 2,
	"PERM_FAILURE": 3,
	"LOOKING":      4,
	"SUPERSEDED":   5,
	"POSTED":       6,
	"REVOKED":      7,
}
```

@maxtaco

### CODE REVIEW

File: src/emit.iced [hunk 1]
```diff
@@ -122,6 +122,15 @@ exports.GoEmitter = class GoEmitter
     @untab()
     @output ")"
 
+    @output "var #{t.name}_map = map[string]#{t.name}{"
```
Comment (maxtaco) [inline src/emit.iced]: maybe be more Go-y:

```
#{t.name}Map
```
?

Comment (mlsteele) [inline src/emit.iced]: Ok, the values are like `ProofStatus_NONE`, which is better `NameMap` or `Name_Map`?

**Comment by maxtaco** [comment]
👍

**Comment by mlsteele** [comment]
changelog ok?

**Comment by maxtaco** [comment]
👍 👍

**Pull request merged**